### ISSUE TITLE
CA1001 is not shown for target-typed new expressions

### ISSUE THREAD

**Issue by cremor** [dotnet/roslyn-analyzers#5834]
### Analyzer

**Diagnostic ID**: [CA1001](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1001): `Types that own disposable fields should be disposable`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 6.0.101](https://dotnet.microsoft.com/download/dotnet/6.0)

### Describe the bug
The diagnostic CA1001 is not recognized for fields that are created with a target-typed new expression.

### Steps To Reproduce

1. Create a new .NET 6 class library project.
2. Add this to the csproj file: `<AnalysisMode>All</AnalysisMode>`
3. Add this class:
```cs
    public sealed class AsyncLock
    {
        private readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1, 1);
    }
```
4. Notice that CA1001 is shown -> good
5. Use the quick fix for IDE0090 to simplify the new expression to a target-typed new expression (`private readonly SemaphoreSlim _semaphore = new(1, 1);`).
6. Notice that CA1001 is not shown any more -> bug

### Expected behavior
CA1001 should be shown for target-typed new expressions.

### Actual behavior
CA1001 is not shown for target-typed new expressions.

**Comment by Youssef1313** [comment]
@sharwell This *could* be related to the `IOperation` behavior you recently noticed (https://github.com/dotnet/roslyn-analyzers/pull/5831).

**Comment by Youssef1313** [comment]
Took a quick look now, it's not `IOperation`-related.

https://github.com/dotnet/roslyn-analyzers/blob/7c544ae66c44f1705ef583b112def4db52c0179a/src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CSharpTypesThatOwnDisposableFieldsShouldBeDisposable.cs#L43

https://github.com/dotnet/roslyn-analyzers/blob/7c544ae66c44f1705ef583b112def4db52c0179a/src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CSharpTypesThatOwnDisposableFieldsShouldBeDisposable.cs#L32

The above two lines should use `BaseObjectCreationExpressionSyntax` instead of `ObjectCreationExpressionSyntax`

**Comment by cremor** [comment]
@Youssef1313 Thanks!

So it seems like it would be a good idea to not only fix this analyzer but also check all other usages of `ObjectCreationExpressionSyntax` if they maybe need to be changed.
https://github.com/dotnet/roslyn-analyzers/search?q=ObjectCreationExpressionSyntax

**Comment by Youssef1313** [comment]
@cremor Other usages of `ObjectCreationExpressionSyntax` needs to be carefully examined to determine whether it's a bug or not, and those changes should be supported by unit tests. If you want to contribute a fix here (either for CA1001 only or reviewing all usages), I can help you getting a PR ready.
